components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -4569,4 +4570,40 @@ public static boolean isFederatedRoleBasedAuthzEnabled(String clientId) throws I
         }
         return isFederatedRoleBasedAuthzEnabled;
     }
+
+    public static String[] getRequestedOIDCScopes(String[] requestedScopes)
+            throws IdentityOAuthAdminException {
+        if (requestedScopes == null) {
+            throw new IdentityOAuthAdminException("Requested scopes are null");
+        }
+        String[] oidcScopes =
+                OAuth2ServiceComponentHolder.getInstance().getOAuthAdminService().getScopeNames();
+        if (oidcScopes == null) {
+            throw new IdentityOAuthAdminException("Error occurred during getting OIDC scopes");
+        }
+        Set<String> oidcScopeSet = new HashSet<>(Arrays.asList(oidcScopes));
+        List<String> requestedOIDCScopes = Arrays.stream(requestedScopes).distinct()
+                .filter(s->oidcScopeSet.contains(s)).collect(Collectors.toList());
+        return requestedOIDCScopes.toArray(new String[requestedOIDCScopes.size()]);
+    }
+
+    public static String[] removeOIDCScopesFromRequestedScopes(String[] requestedScopes)
```

Comment:
Consider whether throwing exceptions is required for empty requested scopes and oidc scopes cases.
Also see if we could return quickly if the oidc scopes list is empty

Reply:
fixed